frontend/src/components/ShootDns/GDnsProviderRow.vue

```diff
@@ -176,58 +164,85 @@ export default {
-        ? 'Primary Provider type cannot be changed after cluster creation'
-        : ''
-    },
-    primaryReadonly () {
-      return !this.isNewCluster && this.primary
-    },
     dnsProviderType: {
       get () {
-        return this.dnsProvider.type
+        return this.dnsProvider?.type
       },
       set (value) {
         this.dnsProvider.type = value
         const dnsSecrets = this.dnsSecretsByProviderKind(value)
         const defaultDnsSecret = head(dnsSecrets)
-        this.dnsProvider.secretName = get(defaultDnsSecret, 'metadata.secretRef.name', null)
+        this.extensionDnsProviderSecret = defaultDnsSecret
       },
     },
-    dnsProviderSecret: {
+    extensionDnsProviderSecret: {
       get () {
-        return find(this.dnsSecrets, ['metadata.secretRef.name', this.dnsProvider.secretName])
+        const resourceName = this.dnsProvider.secretName
+        const secretName = this.getResourceRefName(resourceName)
+        return find(this.dnsSecrets, ['metadata.secretRef.name', secretName])
       },
       set (value) {
-        this.dnsProvider.secretName = get(value, 'metadata.secretRef.name', null)
+        this.deleteResource(this.dnsProvider.secretName)
+        const secretName = get(value, 'metadata.secretRef.name')
+        const resourceName = this.getDnsServiceExtensionResourceName(secretName)
+        this.dnsProvider.secretName = resourceName
+        this.setResource({
+          name: resourceName,
+          resourceRef: {
+            apiVersion: 'v1',
+            kind: 'Secret',
+            name: secretName,
+          },
+        })
       },
     },
-  },
-  watch: {
-    'v$.$invalid' (value) {
-      const oldValue = !this.dnsProvider?.valid
-      if (oldValue !== value && !this.dnsProvider.readonly) {
-        this.dnsProvider.valid = !value
-      }
+    excludeDomains: {
+      get () {
+        return get(this.dnsProvider, 'domains.exclude')
+      },
+      set (value) {
+        set(this.dnsProvider, 'domains.exclude', value)
+      },
+    },
+    includeDomains: {
+      get () {
+        return get(this.dnsProvider, 'domains.include')
+      },
+      set (value) {
+        set(this.dnsProvider, 'domains.include', value)
+      },
+    },
+    excludeZones: {
+      get () {
+        return get(this.dnsProvider, 'zones.exclude')
+      },
+      set (value) {
+        set(this.dnsProvider, 'zones.exclude', value)
+      },
+    },
+    includeZones: {
+      get () {
+        return get(this.dnsProvider, 'zones.include')
+      },
+      set (value) {
+        set(this.dnsProvider, 'zones.include', value)
+      },
+    },
+    filterSecretNames () {
+      return this.dnsServiceExtensionProviders.map(provider => {
+        const secretName = this.getResourceRefName(provider.secretName) // provider.secretName is the resource name
+        if (secretName !== this.extensionDnsProviderSecret?.metadata.name) {
+          return secretName
+        }
+        return undefined
+      })
```

Comment:
```suggestion
      const dnsProviderSecretName = get(this.dnsProviderSecret, 'metadata.name') 
      return this.dnsServiceExtensionProviders.filter(provider => {
        return dnsProviderSecretName !== this.getResourceRefName(provider.secretName)) {
      })
```

Reply:
this will not work... same problem as mentioned above

Reply:
Same problem as mentioned above? I don't get it. Why do we map something to undefined?

Reply:
Sorry, you are right. Not the same problem. The optimized code should look like this:
```
const dnsProviderSecretName = get(this.dnsServiceExtensionProviderSecret, 'metadata.name')
const secretNames = this.dnsServiceExtensionProviders.map(({ secretName }) => this.getResourceRefName(secretName))
const filtered = secretNames.filter(secretName => {
  return dnsProviderSecretName !== secretName
})
return filtered
```